src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -334,8 +334,14 @@ private async Task RefreshIndividualKeyValues()
                         // Check if a change has been detected in the key-value registered for refresh
                         if (keyValueChange.ChangeType != KeyValueChangeType.None)
                         {
-                            ProcessChanges(Enumerable.Repeat(keyValueChange, 1));
+                            if (changeWatcher.RefreshAll)
+                            {
+                                shouldRefreshAll = true;
+                                break;
+                            }
+
                             hasChanged = true;
+                            ProcessChanges(Enumerable.Repeat(keyValueChange, 1));

```

Comment:
when do we call ProcessChanges in case of changeWatcher.RefreshAll?

Reply:
`ProcessChanges` is used to incrementally update or delete a setting in the cached configuration. In case of `refreshAll`, we load all configuration from scratch, so we don't need to call `ProcessChanges`.

Reply:
What about if the watch key isn't part of the select filters?

Reply:
`LoadAll()` is called on line 392. `LoadAll()` ensures that all the keys registered for refresh, like sentinel keys, are obtained.

Reply:
Thanks!